src/web100srv.c

```diff
@@ -529,21 +334,23 @@ void cleanup(int signo) {
       count_vars = tcp_stat_init(VarFileName);
 
       // The administrator view automatically generates a usage page for the
-      // NDT server.  This page is then accessable to the general public.
+      // NDT server.  This page is then accessible to the general public.
       // At this point read the existing log file and generate the necessary
       // data.  This data is then updated at the end of each test.
       if (admin_view == 1) view_init(refresh);
       break;
 
     case SIGCHLD:
-      // Set a flag. The flag is checked near the top of the main wait loop, so
-      // it will only be accessed once and only the testing proces will attempt
-      // to do something with it.
-      if (ndtpid != getppid()) {
-        sig17++;
-        log_println(5, "Signal 17 (SIGCHLD) received - completed tests = %d",
-                    sig17);
+      // When a child exits, send a message to signalfd_write that will cause
+      // the server's select() to wake up.  Only send this message in the
```

Comment:
I'm confused by this. Isn't this message read in `read_from_parent` which is the *child's* select? But then how is that select running if we've received a `SIGCHILD`?